src/UCTNode.cpp

```diff
@@ -300,36 +300,47 @@ UCTNode* UCTNode::uct_select_child(int color) {
     auto parentvisits = size_t{0};
     for (const auto& child : m_children) {
         if (child->valid()) {
-            parentvisits += child->get_visits();
-            if (child->get_visits() > 0) {
+            if (!child->first_visit()) {
+                parentvisits += child->get_visits();
                 total_visited_policy += child->get_score();
+            } else if (child->get_score() > best_unvisited_psa) {
+                best_unvisited_psa = child->get_score();
+                best = child.get();
             }
         }
     }
 
-    auto numerator = std::sqrt((double)parentvisits);
-    auto fpu_reduction = cfg_fpu_reduction * std::sqrt(total_visited_policy);
-    // Estimated eval for unknown nodes = original parent NN eval - reduction
-    auto fpu_eval = get_net_eval(color) - fpu_reduction;
+    if (parentvisits > 0) {
```

Comment:
It's cleaner to inverse the condition and exit immediately. This also removes the need to indent the following big block.